src/server/index.ts

```diff
@@ -254,9 +317,42 @@ export class DocumentDriveServer implements IDocumentDriveServer {
                 throw new Error('Invalid Document Drive document');
             }
 
+            // filter out signals that created a document that was deleted by a subsequent signal
```

Comment:
This flow needs some improvements.

Right now we run all document drive operations and their dispatched signals (which create/delete new documents in the database).
At the end of all signals we store the operations.
Only after that do we create the synchronization units.

If the operations are `[ADD_FILE_NODE, DELETE_FILE_NODE]`:
1. Document of the file is created in the storage layer
2. Document of the file is deleted in the storage layer
3. ADD_FILE_NODE and DELETE_FILE_NODE are stored in storage layer
4. `_onDocumentCreated` is called but the document no longer exists in the storage layer because of step `2`
5 `_onDocumentDeleted' is called

Reply:
Yeah  good call. It's the kind of inefficiency that's okay for now, but we should start thinking about it already because it will have a big impact.

I'm getting the impression that we will likely need a set of edge case detection functions that handle situations like this and optimize them.

Maybe we could have some kind of middleware that runs between the end of all signals and storing the operations? Could be a place to leverage the skip mechanism.